docs/source/10_deployment/12_dask.md

```diff
@@ -0,0 +1,313 @@
+
+    def _exists(self) -> bool:
+        return self._name in Client.current().list_datasets()
+
+    def _release(self) -> None:
+        Client.current().unpublish_dataset(self._name)
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(name=self._name)
+
+
+class DaskRunner(AbstractRunner):
+    """``DaskRunner`` is an ``AbstractRunner`` implementation. It can be
+    used to distribute execution of ``Node``s in the ``Pipeline`` across
+    a Dask cluster, taking into account the inter-``Node`` dependencies.
+    """
+
+    def __init__(self, client_args: Dict[str, Any] = {}, is_async: bool = False):
+        """Instantiates the runner by creating a ``distributed.Client``.
+
+        Args:
+            client_args: Arguments to pass to the ``distributed.Client``
+                constructor.
+            is_async: If True, the node inputs and outputs are loaded and saved
+                asynchronously with threads. Defaults to False.
+        """
+        super().__init__(is_async=is_async)
+        Client(**client_args)
+
+    def __del__(self):
+        Client.current().close()
+
+    def create_default_data_set(self, ds_name: str) -> _DaskDataSet:
+        """Factory method for creating the default dataset for the runner.
+
+        Args:
+            ds_name: Name of the missing dataset.
+
+        Returns:
+            An instance of ``_DaskDataSet`` to be used for all
+            unregistered datasets.
+        """
+        return _DaskDataSet(ds_name)
+
+    def _run_node(
+        self,
+        node: Node,
+        catalog: DataCatalog,
+        is_async: bool = False,
+        run_id: str = None,
+        *dependencies: Node,
+    ) -> Node:
+        """Run a single `Node` with inputs from and outputs to the `catalog`.
+
+        Wraps ``run_node`` to accept the set of ``Node``s that this node
+        depends on. When ``dependencies`` are futures, Dask ensures that
+        the upstream node futures are completed before running ``node``.
+
+        Args:
+            node: The ``Node`` to run.
+            catalog: A ``DataCatalog`` containing the node's inputs and outputs.
+            is_async: If True, the node inputs and outputs are loaded and saved
+                asynchronously with threads. Defaults to False.
+            run_id: The id of the pipeline run.
+            dependencies: The upstream ``Node``s to allow Dask to handle
+                dependency tracking. Their values are not actually used.
+
+        Returns:
+            The node argument.
+        """
+        return run_node(node, catalog, is_async, run_id)
+
+    def _run(
+        self, pipeline: Pipeline, catalog: DataCatalog, run_id: str = None
+    ) -> None:
+        nodes = pipeline.nodes
+        load_counts = Counter(chain.from_iterable(n.inputs for n in nodes))
+        node_dependencies = pipeline.node_dependencies
+        node_futures = {}
+
+        client = Client.current()
```

Comment:
[DASK EXPERTISE REQUESTED] If somebody has familiarity with Dask, a review of how I get the `Client` would be very helpful. I think `worker_client` in `_DaskDataSet` is correct, but not sure if I should be using `Client.current()` the way I am in `DaskRunner`. I think `worker_client` is unnecessary here, since it all runs on the scheduler, and `Client.as_current` seems to be for a use case where you have a client object already and want to use it, but I don't find much documentation around this and most of my understanding is from reading the `dask.distributed` source.

Reply:
@deepyaman I would ask for a review on one of these channels
https://docs.dask.org/en/stable/support.html

Reply:
@datajoely I created a thread for the below question; I feel more confident about this one, more I think about it.